ambry-api/src/main/java/com.github.ambry/account/Container.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+
+/**
+ * A representation of a container. A container virtually groups a number of blobs under the same {@link Account},
+ * so that an operation on the container will be applied to all the blobs of the container. There can be multiple
+ * containers under the the same {@link Account}, but containers cannot be nested. Eventually, one and only one
+ * container needs to be specified when posting a blob. Container id is part of the blob’s properties, and cannot
+ * be modified once the blob is created.
+ *
+ * Container name is provided by a user as an external reference to that container under an {@link Account}. Container
+ * id is one-to-one mapped to container name, and serves as an internal identifier of a container. Container name/id
+ * has to be distinct within the same {@link Account}, but can be the same across different {@link Account}s. Container
+ * metadata is made in JSON, which is generic to contain additional information of the metadata.
+ *
+ * Version 1 of Container metadata in JSON is in the format below.
+ *  {
+ *    "containerName": "MyPrivateContainer",
+ *    "description": "This is my private container",
+ *    "isPrivate": "true",
+ *    "containerId": 0,
+ *    "version": 1,
+ *    "status": "active"
+ *  }
+ */
+public class Container {
+  // static variables
+  public static final String CONTAINER_METADATA_VERSION_KEY = "version";
+  public static final String CONTAINER_NAME_KEY = "containerName";
+  public static final String CONTAINER_ID_KEY = "containerId";
+  public static final String CONTAINER_STATUS_KEY = "status";
+  public static final String CONTAINER_DESCRIPTION_KEY = "description";
+  public static final String CONTAINER_IS_PRIVATE_KEY = "isPrivate";
+  public static final String CONTAINER_STATUS_ACTIVE = "active";
```

Comment:
again, can we have an enum for the states. Also, do add description as to what each state means